apps/frontend/src/app/Database/DataManagers/ArtifactData.ts

```diff
@@ -128,15 +112,22 @@ export class ArtifactDataManager extends DataManager<string, "artifacts", ICache
           }
           isUpgrade ? result.artifacts.upgraded.push(art) : result.artifacts.unchanged.push(art)
           idsToRemove.delete(match.id)
-          if (importId) {
-            delete this.data[match.id]
-            this.removeStorageEntry(match.id)
-          } else importId = match.id
+          if (importId) this.remove(match.id, false)// Do not notify, since this is a "replacement"
+          else importId = match.id
           importArt = { ...art, location: hasEquipment ? art.location : match.location }
         }
       }
       if (importId) {
-        swapId(importId, generateRandomArtID(takenIds))
+        if (this.get(importId)) { // swap existing to another id
```

Comment:
Why do we need to swap

Reply:
import artifact wants to be saved to `importId`, however there is already an artifact that exists in the database. This swap swaps the existing database to a new id so `importId` is empty for the import artifact

Reply:
```suggestion
        if (this.get(importId)) { // importid already in use, get a new id
```